extensions/amp-story-interactive/0.1/amp-story-interactive-img-poll.js

```diff
@@ -37,7 +61,53 @@ export class AmpStoryInteractiveImgPoll extends AmpStoryInteractive {
 
   /** @override */
   buildComponent() {
-    this.rootEl_ = htmlFor(this.element)`<p>Image poll component</p>`;
+    this.rootEl_ = buildImgTemplate(this.element);
+    this.attachContent_(this.rootEl_);
     return this.rootEl_;
   }
+
+  /**
+   * Finds the prompt and options content
+   * and adds it to the poll element.
+   *
+   * @private
+   * @param {Element} root
+   */
+  attachContent_(root) {
+    this.attachPrompt_(root);
+
+    const optionContainer = this.rootEl_.querySelector(
+      '.i-amphtml-story-interactive-img-option-container'
+    );
+    this.options_.forEach((option, index) =>
+      optionContainer.appendChild(this.configureOption_(option, index))
+    );
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the poll element.
+   *
+   * @param {!./amp-story-interactive-abstract.OptionConfigType} option
+   * @return {!Element}
+   * @private
+   */
+  configureOption_(option) {
+    const convertedOption = buildOptionTemplate(this.element);
+    convertedOption.optionIndex_ = option['optionIndex'];
+
+    // Extract and structure the option information
+    setStyle(
+      convertedOption.querySelector(
+        '.i-amphtml-story-interactive-img-option-img'
+      ),
+      'background-image',
+      'url(' + option['image'] + ')'
```

Comment:
Please add a TODO and/or file a ticket to rewrite this image URL (if it's a URL, publishers could also pass a base64 inline image) to an AMP Cache URL for cached documents. Matias should be able to explain what this means :))

Reply:
I'll push this along with the changes Matias requested